src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -717,15 +721,39 @@ public InstancesInfo getApplicationInstances(CloudApplication application) {
     public List<CloudApplication> getApplications() {
         List<CloudApplication> applications = fetchListWithAuxiliaryContent(this::getApplicationResources,
                                                                             this::zipWithAuxiliaryApplicationContent);
-        return addMetadataIfNotEmpty(applications);
+        return addMetadataIfNotEmpty(addStacksToCloudApplications(applications));
+    }
+
+    private List<CloudApplication> addStacksToCloudApplications(List<CloudApplication> cloudApps) {
+        return cloudApps.stream()
+                        .map(this::addStackToCloudApplication)
+                        .collect(Collectors.toList());
+    }
+
+    private CloudApplication addStackToCloudApplication(CloudApplication application) {
+        UUID cloudStackId = application.getStaging()
+                                       .getStackId();
+        CloudStack cloudStack = retrieveCloudStack(cloudStackId, () -> ImmutableRawCloudStack.of(getStackResource(cloudStackId).block())
+                                                                                             .derive());
+        return copyCloudApplicationWithStacks(application, cloudStack).derive();
+    }
+
+    private CloudStack retrieveCloudStack(UUID rawCloudStackId, Supplier<CloudStack> cloudStack) {
```

Comment:
The name of the CF entity is just `Stack`. We don't need to call it by our wrapper object's name `CloudStack`.
Another thing, the name of the second parameter isn't descriptive of what it is.